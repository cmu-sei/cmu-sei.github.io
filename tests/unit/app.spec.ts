import { shallowMount } from '@vue/test-utils'
import { mockNuxtImport } from 'nuxt-vitest/utils'
import { describe, expect, it, vi } from 'vitest'
import Component from '../../app.vue'

mockNuxtImport('useGithub', () => {
  return () => {
    return {
      api: vi.fn((name, type) => {
        return {
          data: {
            owner: {
              login: name.split('/')[0],
              html_url: 'https://<OWNER_URL>'
            },
            name: name.split('/')[1],
            html_url: 'https://<URL>',
            description: `Type: ${type}`,
            stargazers_count: 20,
            forks_count: 54,
            language: 'Java'
          }
        }
      }),
      colors: {
        'Java': {
          'color': '#814CCC',
          'url': '<MOCK_URL>'
        }
      },
      repos: [
        {
          name: 'test/hello',
          type: 'repo'
        },
        {
          name: 'test2/world',
          type: 'repo'
        }
      ]
    }
  }
})

const mocks = {
  $ga: {
    page: vi.fn()
  }
}

describe('App.vue', () => {
  it('gets all and recent repos on mounted', async () => {
    const wrapper = shallowMount(Component)
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot('homepage')
  })

  // it('throws error when repos are unable to be retrieved', async () => {
  //   github.mockImplementationOnce(() => Promise.reject(Error('error thrown')))
  //   const wrapper = shallowMount(Component, { mocks })
  //   await wrapper.vm.$nextTick()
  //   expect(wrapper.vm.error).toBe(true)
  // })

  it('matches snapshots', async () => {
    const wrapper = shallowMount(Component, { mocks })
    expect(wrapper.html()).toMatchSnapshot('Loading')
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot('DoneLoading')
  })
})
