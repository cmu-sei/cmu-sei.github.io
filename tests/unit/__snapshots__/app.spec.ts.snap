// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > gets all and recent repos on mounted > homepage 1`] = `
"<div class="flex flex-col h-screen w-screen">
  <title-stub></title-stub>
  <body-stub class="bg-white text-gray-900"></body-stub>
  <anonymous-stub></anonymous-stub>
  <anonymous-stub class="shrink-0"></anonymous-stub>
  <main class="grow mb-2">
    <anonymous-stub></anonymous-stub>
    <div>
      <anonymous-stub></anonymous-stub>
    </div>
  </main>
  <site-footer-stub class="shrink-0"></site-footer-stub>
</div>"
`;

exports[`App.vue > matches snapshots > DoneLoading 1`] = `
"<div class="flex flex-col h-screen w-screen">
  <title-stub></title-stub>
  <body-stub class="bg-white text-gray-900"></body-stub>
  <anonymous-stub></anonymous-stub>
  <anonymous-stub class="shrink-0"></anonymous-stub>
  <main class="grow mb-2">
    <anonymous-stub></anonymous-stub>
    <div>
      <anonymous-stub></anonymous-stub>
    </div>
  </main>
  <site-footer-stub class="shrink-0"></site-footer-stub>
</div>"
`;

exports[`App.vue > matches snapshots > Loading 1`] = `
"<div class="flex flex-col h-screen w-screen">
  <title-stub></title-stub>
  <body-stub class="bg-white text-gray-900"></body-stub>
  <anonymous-stub></anonymous-stub>
  <anonymous-stub class="shrink-0"></anonymous-stub>
  <main class="grow mb-2">
    <anonymous-stub></anonymous-stub>
    <div class="bg-gray-100">
      <!--v-if-->
    </div>
    <!--v-if-->
  </main>
  <site-footer-stub class="shrink-0"></site-footer-stub>
</div>"
`;
